### DISCUSSION TITLE
FailedPreconditionError: Attempting to use uninitialized value dense_22/bias

### DISCUSSION THREAD

**Discussion by dpde101** [lululxvi/deepxde#677] [Q&A] [answered]
hi i see your reply of yesterday to use `X = np.array([[3]])` for model.predict but when i do i have error 

`FailedPreconditionError: Attempting to use uninitialized value dense_22/bias`

is complete erreur message

```
FailedPreconditionError                   Traceback (most recent call last)

/tensorflow-1.15.0/python3.6/tensorflow_core/python/client/session.py in _do_call(self, fn, *args)
   1364     try:
-> 1365       return fn(*args)
   1366     except errors.OpError as e:

9 frames

FailedPreconditionError: Attempting to use uninitialized value dense_22/bias
	 [[{{node dense_22/bias/read}}]]


During handling of the above exception, another exception occurred:

FailedPreconditionError                   Traceback (most recent call last)

/tensorflow-1.15.0/python3.6/tensorflow_core/python/client/session.py in _do_call(self, fn, *args)
   1382                     '\nsession_config.graph_options.rewrite_options.'
   1383                     'disable_meta_optimizer = True')
-> 1384       raise type(e)(node_def, op, message)
   1385 
   1386   def _extend_graph(self):

FailedPreconditionError: Attempting to use uninitialized value dense_22/bias
	 [[node dense_22/bias/read (defined at /tensorflow-1.15.0/python3.6/tensorflow_core/python/framework/ops.py:1748) ]]

Original stack trace for 'dense_22/bias/read':
```

**Comment by dpde101** [comment]
and you can pls give me notation for initial value for exemple dy/dt=f(t,y) and y(0)=1 how to writ initial condition not boundary condition thanks

**Reply by PatrikDeath** [reply]
Já to věděl že jseš zasranej hacker

**Comment by lululxvi** [comment]
For the error, did you initialize/train the network before prediction?

The IVP can be viewed as BVP where the boundary is at left point. check #674

**Comment by dpde101** [comment]
i get error when i initialize/train before and after prediction both. for initial value is there way to declare it without putting the point in the geometry like boundary value ? thank you greetings

**Comment by lululxvi** [comment]
You can only make prediction after the network is initialized. If you don't want to do the training, you can just call train with 0 steps.

DeepXDE  supports IC for at least 1D problem, e.g., Burgers https://github.com/lululxvi/deepxde/blob/master/examples/Burgers.py . But DeepXDE currently doesn't support IC for 0D problem, because it is just a special case of left boundary problem.

**Comment by dpde101** [comment]
i think it is because i try to use model.predict(X) in arithmetic
operations (+ and -) do you have any idea how to bypass this thanks

**Comment by lululxvi** [comment]
I am not sure what you mean. Could you show me the code?

**Comment by dpde101** [comment]
```
    layer_size = [1] + [35] * 3 + [1]
    activation = "tanh"
    initializer = "Glorot uniform"
    geom = dde.geometry.Interval(1, 20)
    net = dde.maps.FNN(layer_size, activation, initializer)

    t0=0
    i=0
    X=np.array([[i+1]])




    while (i<20):
           bc = dde.DirichletBC(geom, lambda x:0*x+t0, boundary)
           data = dde.data.PDE(geom, 1, ode_system, bc, 30, 2, num_test=100)
           model = dde.Model(data, net)
           t0=(t0+model.predict(x))
           model.compile("adam", lr=0.001, metrics=["l2 relative error"])
           losshistory, train_state = model.train(epochs=20000); 
           i=i+1

    dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

edit: i still get same error after putting  `t0=(t0+model.predict(x))` after model.train so i think issues is in model.predict

**Comment by lululxvi** [comment, marked as answer]
Because you predict before training.

**Comment by dpde101** [comment]
absolutely correct, i make mistake early in the code i forget copy paste a
line of code

thank s

**Comment by lululxvi** [comment]
@dpde101 In fact I find that DeepXDE supports using IC for IVP, see https://github.com/lululxvi/deepxde/blob/master/examples/ode_system.py

**Comment by dpde101** [comment]
thanks for tell me

can you give example for syntax for example dy1/dt=f1(t,y), dy2/dt=f2(t,y)
and y1(0)=1 and y2(0)=3 what are the expressions for boundary and ic1 and
ic2 and data

thank you

**Comment by lululxvi** [comment]
I am not sure what you mean. The ODE system example should have what you need.

**Comment by dpde101** [comment]
not really because they use a function (np.sin and np.cos) to define ic1
and ic2 but i want to use lambda x

this is exctly like https://github.com/lululxvi/deepxde/discussions/673 where you
gave a well detailed solution
https://github.com/lululxvi/deepxde/discussions/673#discussioncomment-2748875, what
i want is the same thing but for initial conditions instead of boundary
conditions

thankyou

**Comment by lululxvi** [comment]
- `np.sin` is the same as lambda expression, right? This is all about Python grammar.
- IC and BC have almost the same usage. 
- You can find more examples, e.g., https://github.com/lululxvi/deepxde/blob/master/examples/Lorenz_inverse.py

**Comment by athtareq** [comment]
I have been following this, thanks for the reply

in example ic1 = dde.IC(geom, lambda X: -8 * np.ones(X.shape), boundary,
component=0) , is this referring to the IC:  y1(-8)=0 or y1(0)=-8 ?

**Comment by dpde101** [comment]
@tareqath haha you are following without telling me wach asahbi hetta nta
kheddam hna katsowel, khasna nemchi l Casablanca chi nhar

@luluxvi thankyou smuch i understand it now

**Comment by lululxvi** [comment]
@tareqath 
- `lambda X: -8 * np.ones(X.shape)` is `f(x) = -8`
- `component=0` means that this IC/BC is for the first component, i.e., `y(0)`

So, it is `y(0) = -8`.

**Comment by athtareq** [comment]
I can't stress this enough: thank you so much !
